File: php/Queue.php
<!-- ###################################
##### PHP Queue Data Structure
###################################
##### July 4th, 2020
################################### -->


<?php

class Queue {
    public function __construct(){
        $this->data = [];
    }

    public function enter ($new) {
        $this->data[] = $new;
    }

    public function leave(){
       array_shift($this->data); 
    }

    public function current(){
        return $this->data;
    }
}

$myLine = new Queue();

$myLine->enter(1);
$myLine->enter(2);
$myLine->enter(3);
var_dump($myLine->current());
$myLine->leave();
var_dump($myLine->current());

?>

File: go/Queue.go
///////////////////////////////////
// Queue Data Structure
//////////////////////////////////
// 7-7-2020

package main

import "fmt"

//Create Queue Struct
type Queue struct {
	data []int
}

//Queue Methods, pointer methods so it mutates the Queue
func (q *Queue) enter(n int){
	q.data = append(q.data, n)
}

func (q *Queue) leave(){
	q.data = q.data[1:]
}

func main(){
	var myLine = Queue{[]int{}}
	myLine.enter(1)
	myLine.enter(2)
	myLine.enter(3)
	fmt.Println(myLine.data)
	myLine.leave()
	fmt.Println(myLine.data)
	
}
